### ISSUE TITLE
Migrate code to .NET Core

### ISSUE THREAD

**Issue by conniey** [sillsdev/icu-dotnet#8]
A good enhancement would be if this code compiled on [.NET Core](https://dotnet.github.io/) so the library would run cross-platform.

I've already done work to migrate the code to run on [.NETStandard1.3](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md) but I wanted your feedback and criteria if your team wanted this feature in your repository

**Comment by JohnThomson** [comment]
Hi Connie,

Your enhancements sound useful. I am trying to track down who (if anyone)
on our team is currently responsible for this project to get a definite
response for you.

Thanks for your interest in contributing!

John Thomson

**Comment by ermshiperete** [comment]
We're already running icu-dotnet cross-platform (with mono 3.x). Since a lot of our applications make use of Winforms I don't think we can currently switch to .NET Core for the master branch, but I'd be more than happy to see it on a separate branch. That way projects that don't require Winforms and don't rely on Mono 3.x could start using it. Would be great.

**Comment by JohnThomson** [comment]
Why does our use of Winforms prevent moving to .NET core? I don't know
exactly what that is, but I would not have expected much if any UI in
icu.net.

JohnT

On Mon, May 30, 2016 at 5:34 AM, Eberhard Beilharz <notifications@github.com

**Comment by ermshiperete** [comment]
@JohnThomson Yes, icu.net doesn't have any UI nor depend on Winforms (I think :smile: ). 

[.NET Core](https://en.wikipedia.org/wiki/.NET_Framework#.NET_Core) (CoreCLR) is a subset of the .NET framework that Microsoft released under the MIT license. Among other things it is missing Winforms. And it has additional APIs that are not found in the regular .NET framework.

I haven't played around with it, so I don't know if it is possible to compile against .NET Core but then run with .NET 4.5 or Mono 3.x or if that causes problems. If that works then I don't see a problem, but I don't want to have to install both Mono 3.x **and** .NET Core for an application to run on Linux.

**Comment by conniey** [comment]
>  I don't know if it is possible to compile against .NET Core but then run with .NET 4.5 or Mono 3.x or if that causes problems.

According to [.NETStandard to Platform mapping](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md#mapping-the-net-platform-standard-to-platforms), if you target your library for .NETStandard, Version=1.1, it will run on .NET Framework v4.5.  Since mono runs on the full .NET Framework, it shouldn't be an issue.
